server/src/controllers/employer/getAllEmployers.js

```diff
@@ -25,26 +25,57 @@ const getAllEmployersRequestHandler = async (req, res) => {
       query.name = { [Op.like]: `%${employerName}%` };
     }
 
-    let employerContactIds = [];
+    let employerContacts = [];
     if (contactName) {
-      const employerContacts = await EmployerContact.findAll({
+      employerContacts = await EmployerContact.findAll({
         where: {
           name: { [Op.like]: `%${contactName}%` },
         },
       });
-      employerContactIds = employerContacts.map((contact) => contact.id);
+      logger.info(employerContacts)
     }
 
-    logger.info(`Employer contact ids: ${employerContactIds}`);
-    logger.info(`Employer Name: ${contactName}`);
-
-    if (employerContactIds.length > 0) {
-      query.employer_contact_id = { [Op.in]: employerContactIds };
+    if (phoneNumber) {
+      // Validate that phoneNumber only contains digits
+      if (/^\d+$/.test(phoneNumber)) {
+        // Checking with Employer Table
+        const employerPhoneNumber = await Employer.findAll({
+          where: {
+            phoneNumber: Sequelize.literal(
+              `REGEXP_REPLACE(phone_number, '[^0-9]', '') REGEXP '${phoneNumber}'`
+            ),
+          },
+        });
+
+        // Checking with EmployerContact Table
+        const phoneNumberContacts = await EmployerContact.findAll({
+          where: {
+            [Op.or]: [
+              {
+                phone_number: Sequelize.literal(
+                  `REGEXP_REPLACE(phone_number, '[^0-9]', '') REGEXP '${phoneNumber}'`
+                ),
+              },
+              {
+                alt_phone_number: Sequelize.literal(
+                  `REGEXP_REPLACE(alt_phone_number, '[^0-9]', '') REGEXP '${phoneNumber}'`
+                ),
+              },
+            ],
+          },
+        });
+        employerContacts = [...employerPhoneNumber, ...phoneNumberContacts, ...employerContacts];
```

Comment:
`employerPhoneNumber` probably shouldn't be added to employer contacts since these are employer objects and not employer contact objects. This means extracting the employer later doesn't make sense